src/core/assertions.js

```diff
@@ -92,36 +98,43 @@ exports.assertObjectMatchSpec = (object, spec, exact = false) => {
         const currentValue = _.get(object, field)
         const expectedValue = Cast.value(value)
 
-        switch (matcher) {
-            case 'match':
-            case 'matches':
-                expect(
-                    currentValue,
-                    `Property '${field}' (${currentValue}) does not match '${expectedValue}'`
-                ).to.match(new RegExp(expectedValue))
-                break
+        if (matchRegex.exec(matcher) !== null) {
```

Comment:
Maybe we could just use `indexOf` which is more performant, `if (matcher.indexOf('match') === 0)`